### PR TITLE
ESQL: Update ProjectOperator to release dropped blocks

### PR DESCRIPTION
**Pull request by ChrisHegarty**
This commit updates the ProjectOperator to release dropped blocks.

### CODE REVIEW

**Review by nik9000 — APPROVED** [review]

File: x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/ProjectOperator.java [hunk 1]
```diff
@@ -52,10 +56,16 @@ protected Page process(Page page) {
         Arrays.fill(blocks, null);
         int b = 0;
         int positionCount = page.getPositionCount();
-        for (int i = bs.nextSetBit(0); i >= 0 && i < page.getBlockCount(); i = bs.nextSetBit(i + 1)) {
+        List<Releasable> blocksToRelease = new ArrayList<>();
+        for (int i = 0; i >= 0 && i < page.getBlockCount(); i++) {
```
Comment (dnhatn) [inline x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/ProjectOperator.java:60]: nit: leftover `i >= 0`?

**Review by dnhatn — APPROVED** [review]
LGTM.

**Review by costin — APPROVED** [review]
LGTM

File: x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/ProjectOperator.java (hunk 1 above)
Comment (ChrisHegarty) [reply x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/ProjectOperator.java]: D'oh! removed. [659399c](https://github.com/elastic/elasticsearch/pull/99885/commits/659399c22d699f40e9b6778798e53cd197534bd0)

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Pull request merged**